ui/app/ducks/metamask/metamask.js

```diff
@@ -227,6 +199,35 @@ export default function reduceMetamask (state = {}, action) {
         },
       })
 
+    case actionConstants.UPDATE_SEND_TOKEN:
+      const newSend = {
+        ...metamaskState.send,
+        token: action.value,
+      }
+      // erase token-related state when switching back to native currency
+      if (newSend.editingTransactionId && !newSend.token) {
+        const unapprovedTx = newSend?.unapprovedTxs?.[newSend.editingTransactionId] || {}
+        const txParams = unapprovedTx.txParams || {}
+        Object.assign(newSend, {
+          tokenBalance: null,
+          balance: '0',
+          from: unapprovedTx.from || '',
+          unapprovedTxs: {
+            ...newSend.unapprovedTxs,
+            [newSend.editingTransactionId]: {
+              ...unapprovedTx,
+              txParams: {
+                ...txParams,
+                data: '',
+              },
+            },
+          },
+        })
+      }
+      return Object.assign(metamaskState, {
```

Comment:
nit: Not sure whether we have a preference between spreading and `Object.assign`, but if we do, we should probably follow one or the other?

Reply:
Using the spread operator here wouldn't be functionally equivalent; this mutates `metamaskState` rather than constructing a new object. Which is fine here, since `metamaskState` is initialized to a new object at the start of the reducer. The equivalent of `{ ...x, ...y }` would be `Obejct.assign({}, x, y)` (with the empty object as the first parameter).

Ultimately I would prefer our convention to be neither of them, but to use `immer` instead. In the meantime either seems fine, with this being slightly better because it allocates one less unnecessary object 🤷‍♂️